### PR TITLE
Add ImGui support to Android

### PR DESCRIPTION
**Pull request by katemonster33**
#### Summary
Features "Added support for showing ImGui screens to Android"

#### Purpose of change
With #72154 merged, there are no blockers remaining to showing ImGui screens in Android. This PR adds ImGui support to Android.

#### Describe the solution
Removed all preprocessor directives walling off ImGui screens being shown from in Android. Added ImGui to Gradle build.

#### Testing
Tested on my Pixel 7 with Android 14.

#### Additional context
Any PRs that have added ImGui support to other screen would need to be updated after this is merged. Or, if they are merged first, this PR should be updated.

### CODE REVIEW

**Comment by katemonster33** [comment]
Builds: https://github.com/katemonster33/Cataclysm-DDA/releases/download/cdda-experimental-2024-03-12-2138/cdda-android-x32-2024-03-12-2138.apk and https://github.com/katemonster33/Cataclysm-DDA/releases/download/cdda-experimental-2024-03-12-2138/cdda-android-x64-2024-03-12-2138.apk

File: src/cata_imgui.cpp [hunk 1]
```diff
@@ -207,6 +208,43 @@ void cataimgui::client::process_input( void *input )
...
+        case KEY_ENTER:
+            return ImGuiKey_Enter;
+        case KEY_ESCAPE:
+            return ImGuiKey_Escape;
+        default:
+            if( cata_key >= 'a' && cata_key <= 'z' ) {
+                return static_cast<ImGuiKey>( ImGuiKey_A + ( cata_key - 'a' ) );
+            } else if( cata_key >= 'A' && cata_key <= 'Z' ) {
+                return static_cast<ImGuiKey>( ImGuiKey_A + ( cata_key - 'A' ) );
+            } else if( cata_key >= '0' && cata_key <= 'Z' ) {
```
Comment (Qrox) [inline src/cata_imgui.cpp:231]:
```suggestion
            } else if( cata_key >= '0' && cata_key <= '9' ) {
```

**Comment by Qrox** [comment]
I remember there being reports of the game occupying only a fraction of the screen on Android recently. Is it related to the ImGUI change, and if so, does this fix it?

**Comment by katemonster33** [comment]
There are reports of that, so far the only thing we know is that it's related to the SDL update to v2.30.0, we haven't gotten enough testing data to determine what we can even do about it. I can't reproduce it and I seem to be the only one working on Android right now.

This PR does not address it.

**Pull request merged**